integration_tests/test_mempool.py

```diff
@@ -24,50 +26,75 @@ def cronos_mempool(tmp_path_factory):
-    assert filter.get_new_entries() == []
+    max_retry = 3
+    for x in range(max_retry):
+        w3: Web3 = cronos_mempool.w3
+        filter = w3.eth.filter("pending")
+        assert filter.get_new_entries() == []
 
-    cli = cronos_mempool.cosmos_cli(0)
-    # test contract
-    wait_for_new_blocks(cli, 1)
-    block_num_2 = w3.eth.get_block_number()
-    print(f"block number contract begin at height: {block_num_2}")
-    contract = deploy_contract(w3, CONTRACTS["Greeter"])
-    tx = contract.functions.setGreeting("world").buildTransaction()
-    signed = sign_transaction(w3, tx)
-    txhash = w3.eth.send_raw_transaction(signed.rawTransaction)
-    w3.eth.wait_for_transaction_receipt(txhash)
-    # check tx in mempool
-    new_txs = filter.get_new_entries()
-    assert txhash in new_txs
+        cli = cronos_mempool.cosmos_cli(0)
+        # test contract
+        wait_for_new_blocks(cli, 1, sleep=0.1)
+        block_num_2 = w3.eth.get_block_number()
+        print(f"block number contract begin at height: {block_num_2}")
+        contract = deploy_contract(w3, CONTRACTS["Greeter"])
+        tx = contract.functions.setGreeting("world").buildTransaction()
+        signed = sign_transaction(w3, tx)
+        txhash = w3.eth.send_raw_transaction(signed.rawTransaction)
+        w3.eth.wait_for_transaction_receipt(txhash)
+        # check tx in mempool
+        new_txs = filter.get_new_entries()
+        assert txhash in new_txs
 
-    greeter_call_result = contract.caller.greet()
-    assert "world" == greeter_call_result
+        greeter_call_result = contract.caller.greet()
+        assert "world" == greeter_call_result
 
-    # check mempool
-    all_pending = w3.eth.get_filter_changes(filter.filter_id)
-    print(f"all pending tx hash after block: {all_pending}")
-    assert len(all_pending) == 0
+        # check mempool
+        all_pending = w3.eth.get_filter_changes(filter.filter_id)
+        print(f"all pending tx hash after block: {all_pending}")
+        assert len(all_pending) == 0
 
-    to = ADDRS["community"]
-    params = {"gasPrice": w3.eth.gas_price}
-    block_num_0, sended_hash_set = send_txs(w3, cli, to, KEYS.values(), params)
+        to = ADDRS["community"]
+        params = {"gasPrice": w3.eth.gas_price}
+        block_num_0, sended_hash_set = send_txs(w3, cli, to, KEYS.values(), params)
 
-    all_pending = w3.eth.get_filter_changes(filter.filter_id)
-    assert len(all_pending) == 0
+        all_pending = w3.eth.get_filter_changes(filter.filter_id)
+        assert len(all_pending) == 0
 
-    block_num_1 = w3.eth.get_block_number()
-    assert block_num_1 == block_num_0, "new block is generated, fail"
-    print(f"all send tx hash: f{sended_hash_set}")
+        block_num_1 = w3.eth.get_block_number()
+        try:
+            assert block_num_1 == block_num_0, "new block is generated"
+            print(f"all send tx hash: f{sended_hash_set}")
+        except AssertionError as e:
+            if x < max_retry - 1:
```

Comment:
this means at most 2 times retry, is this intended?

Reply:
3: 0, 1, 2